pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -242,9 +242,20 @@ func (ssr *Snapshotter) stop() {
 		ssr.deltaSnapshotTimer = nil
 	}
 	ssr.closeEtcdClient()
+}
 
+// SetSnapshotterInactive set the snapshotter state to Inactive.
+func (ssr *Snapshotter) SetSnapshotterInactive() {
+	ssr.SsrStateMutex.Lock()
+	defer ssr.SsrStateMutex.Unlock()
 	ssr.SsrState = brtypes.SnapshotterInactive
-	ssr.SsrStateMutex.Unlock()
+}
+
+// SetSnapshotterActive set the snapshotter state to active.
+func (ssr *Snapshotter) SetSnapshotterActive() {
+	ssr.SsrStateMutex.Lock()
+	defer ssr.SsrStateMutex.Unlock()
```

Comment:
instead of using mutex to change a field in a struct you should perhaps use `atomic.Bool.CompareAndSwap` instead. Usage of `sync.Mutex`

Reply:
I think atomic operations are costly operations as it involves system call, so if we can achieve something without kernel involvement then why not use it, that's why I used mutex instead of atomic ops. WDYT?

Reply:
https://www.sixt.tech/golangs-atomic - atomics are supported by native cpu architectures and are generally faster as compared to mutex. They do not completely replace mutex as such but there is an overlap.